packages/comms/src/ecl/dfuWorkunit.ts

```diff
@@ -0,0 +1,329 @@
+        const service = new FileSprayService(server);
+        return service.SprayVariableEx({
+            ...request
+        }).then(response => {
+            const wuid = response.wuid;
+            return service.GetDFUWorkunit({ wuid }).then(response => {
+                return DFUWorkunit.attach(server, wuid, response.result);
+            });
+        });
+    }
+
+    static despray(server: IOptions | IConnection, request: Partial<FileSpray.Despray>): Promise<DFUWorkunit> {
+        const service = new FileSprayService(server);
+        return service.DesprayEx({
+            ...request
+        }).then(response => {
+            const wuid = response.wuid;
+            return service.GetDFUWorkunit({ wuid }).then(response => {
+                return DFUWorkunit.attach(server, wuid, response.result);
+            });
+        });
+    }
+
+    isComplete(): boolean {
+        switch (this.State) {
+            case States.Finished:
+            case States.Failed:
+            case States.Aborted:
+            case States.NotFound:
+                return true;
+            default:
+        }
+        return false;
+    }
+
+    isFailed() {
+        if (this.isComplete() && this.State !== States.Finished) {
+            return true;
+        }
+        return false;
+    }
+
+    isDeleted() {
+        switch (this.State) {
+            case States.NotFound:
+                return true;
+            default:
+        }
+        return false;
+    }
+
+    isRunning(): boolean {
+        switch (this.State) {
+            case States.Started:
+            case States.Aborting:
+            case States.Queued:
+                return true;
+            default:
+        }
+        return false;
+    }
+
+    async refreshState(): Promise<this> {
+        await this.GetDFUWorkunit();
+        return this;
+    }
+
+    async refreshInfo(): Promise<this> {
+        await this.GetDFUWorkunit();
+        return this;
+    }
+
+    async refresh(full: boolean = false): Promise<this> {
+        if (full) {
+            await this.refreshInfo();
+        } else {
+            await this.refreshState();
+        }
+        return this;
+    }
+
+    //  Monitoring  ---
+    protected _monitor(): void {
+        if (this.isComplete()) {
+            this._monitorTickCount = 0;
+            return;
+        }
+        super._monitor();
+    }
+
+    protected _monitorTimeoutDuraction(): number {
+        const retVal = super._monitorTimeoutDuraction();
+        if (this._monitorTickCount <= 1) {          //  Once
+            return 1000;
+        } else if (this._monitorTickCount <= 3) {   //  Twice
+            return 3000;
+        } else if (this._monitorTickCount <= 5) {   //  Twice
+            return 5000;
+        } else if (this._monitorTickCount <= 7) {   //  Twice
+            return 10000;
+        }
+        return retVal;
```

Comment:
I see there is a typo in "_monitorTimeoutDuraction"...  Also for spray jobs a less aggressive polling scheduling would be more appropriate?

Reply:
I renamed the symbol everywhere, so there are a couple of lines changed in workunit.ts & stateful.ts as well